public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx

```diff
@@ -373,8 +377,27 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
     this.props.panel.updateFieldConfig(config);
   };
 
+  logPanelInfoOnError(error: Error) {
+    const logObj = {
+      error: error.message,
+      options: JSON.stringify(this.props.panel.getOptions()),
+      fieldConfig: JSON.stringify(this.props.panel.fieldConfig.defaults),
+      overrides: JSON.stringify(this.props.panel.fieldConfig.overrides),
+      panelId: String(this.props.panel.id),
+      panelType: this.props.panel.type,
+      panelTitle: this.props.panel.title,
+    };
+
+    faro.api.pushEvent(PanelLogEvents.PANEL_ERROR, logObj);
+  }
+
   onPanelError = (error: Error) => {
     const errorMessage = error.message || DEFAULT_PLUGIN_ERROR;
+
+    if (config.featureToggles.panelMonitoring) {
+      this.logPanelInfoOnError(error);
```

Comment:
just know that this could cause a flood of these large log events. With a dashboard with say 30 visible panels and refresh every 5s and something cause them all to fail it would cause 360 log messages per minute for every user having that dashboard open :)

Reply:
I will need to look into Faro protection, thanks for pointing that out!

Reply:
@torkelo It seems that there is no sampling in Faro yet, although it's planned. To avoid the scenario you mentioned, I am thinking of maybe logging panel crashes only in panel edit? I would assume that if any options related crashes appear, they would mostly pop in panel edit mode. We might potentially lose some opportunities to log in some cases like where we import dashboards with bad panels, but maybe that's the tradeoff?